homeassistant/components/ecobee/climate.py

```diff
@@ -813,6 +850,21 @@ def set_dst_mode(self, dst_enabled):
         """Enable/disable automatic daylight savings time."""
         self.data.ecobee.set_dst_mode(self.thermostat_index, dst_enabled)
 
+    def set_humidifier_mode(self, humidifier_mode):
+        """Set humidifier mode (auto, off, manual)."""
+        if humidifier_mode.lower() not in (self.humidifier_modes):
+            raise ValueError(
+                f"Invalid humidifier_mode value: {humidifier_mode}  Valid values are 'auto', 'off', or 'manual'"
+            )
+
+        if not self.thermostat["settings"]["hasHumidifier"]:
+            raise Exception(
+                "Cannot set humidifier mode since no humidifier is installed."
+            )
```

Comment:
Shouldn't this be the very first guard in this method?

Reply:
Yes it should.